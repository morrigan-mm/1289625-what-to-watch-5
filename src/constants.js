export const PageMovieTab = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const PageType = {
  MOVIE_CARD: `movie-card`,
  USER_PAGE: `user-page`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
