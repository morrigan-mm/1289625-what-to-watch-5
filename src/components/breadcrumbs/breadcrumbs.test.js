import React from "react";
import {StaticRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {AppRoute} from "../../constants";
import Breadcrumbs from "./breadcrumbs";

const breadcrumbs = [
  {
    link: AppRoute.FILM.url({id: `1234`}),
    text: `A Star Is Born`,
  },
  {
    text: `Add Review`
  }
];

describe(`Breadcrumbs`, () => {
  it(`Should render correctly`, () => {
    const tree = renderer
      .create(
          <StaticRouter>
            <Breadcrumbs breadcrumbs={breadcrumbs} />
          </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
