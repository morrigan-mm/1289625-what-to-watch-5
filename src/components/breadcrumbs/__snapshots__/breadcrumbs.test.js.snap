// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Should render correctly 1`] = `
<nav
  className="breadcrumbs"
>
  <ul
    className="breadcrumbs__list"
  >
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
        href="/films/1234"
        onClick={[Function]}
      >
        A Star Is Born
      </a>
    </li>
    <li
      className="breadcrumbs__item"
    >
      <span
        className="breadcrumbs__link"
      >
        Add Review
      </span>
    </li>
  </ul>
</nav>
`;
